tools/RAiDER/delay.py

```diff
@@ -202,6 +204,167 @@ def tropo_delay(args):
 
     return wetDelay, hydroDelay
 
+def tropo_delay_cube(args):
+    """
+    raiderDelay cube generation function.
+    Parameters
+    ----------
+    ll_bounds       - Lat/Lon bbox
+    heights         - Height levels for output cube
+    weather_model   - Type of weather model to use
+    wmLoc           - Directory containing weather model files
+    zref            - max integration height
+    spacinginm      - output cube spacing in meters
+    crs             - output cube pyproj crs
+    out             - output directory
+    outformat       - File format to use for raster outputs
+    time            - Datetime to calculate delay
+    download_only   - Only download the raw weather model data
+    filename        - Output filename
+    verbose         - Verbose printing
+    """
```

Comment:
I agree with this but slightly out of scope for this PR. I'm sticking to the current interface, as I expect this to merge with the single interface once #364 has been resolved.  I'm also a huge fan of `Fire` - which lets me call any python function without having to write a parser interface for testing.  

The prototype currently writes a netcdf file, but will be integrated with `raiderDelay.py` - when it will return only a dataset. Writing gets delegated to `raiderDelay.py`

Reply:
I would be happy to help with something like this - would also help me get started on hyp3 plugin.

Reply:
Also - just a general comment - it would be nice to have python functions that can run corrections outside the CLI - this will be useful debugging too; not to mention for teaching (including myself) purposes.

One of the hardest things about exclusive CLI for modular calls to corrections is that we then write some python function to wrap the CLI via `subprocess.call` which is inevitably harder to catch errors correctly.

Ok, will be quiet.

Reply:
@cmarshak just a note that most of the functionality can in fact be called independently. I think actually the full delay calculation itself may be the only thing that's a tricky because of argument checking, but it could be done with the current setup (see [here](https://github.com/dbekaert/RAiDER/blob/7111fc812fcc1fb8652a3690f07610e989e3a80d/tools/RAiDER/delay.py#L32) for example).